internal/kibana/maintenance_window/models.go

```diff
@@ -0,0 +1,321 @@
+package maintenance_window
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/elastic/terraform-provider-elasticstack/generated/kbapi"
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type MaintenanceWindowModel struct {
+	ID             types.String              `tfsdk:"id"`
+	SpaceID        types.String              `tfsdk:"space_id"`
+	Title          types.String              `tfsdk:"title"`
+	Enabled        types.Bool                `tfsdk:"enabled"`
+	CustomSchedule MaintenanceWindowSchedule `tfsdk:"custom_schedule"`
+	Scope          *MaintenanceWindowScope   `tfsdk:"scope"`
+}
+
+type MaintenanceWindowScope struct {
+	Alerting MaintenanceWindowAlertingScope `tfsdk:"alerting"`
+}
+
+type MaintenanceWindowAlertingScope struct {
+	Kql types.String `tfsdk:"kql"`
+}
+
+type MaintenanceWindowSchedule struct {
+	Start     types.String                        `tfsdk:"start"`
+	Duration  types.String                        `tfsdk:"duration"`
+	Timezone  types.String                        `tfsdk:"timezone"`
+	Recurring *MaintenanceWindowScheduleRecurring `tfsdk:"recurring"`
+}
+
+type MaintenanceWindowScheduleRecurring struct {
+	End         types.String `tfsdk:"end"`
+	Every       types.String `tfsdk:"every"`
+	Occurrences types.Int32  `tfsdk:"occurrences"`
+	OnWeekDay   types.List   `tfsdk:"on_week_day"`
+	OnMonthDay  types.List   `tfsdk:"on_month_day"`
+	OnMonth     types.List   `tfsdk:"on_month"`
+}
+
+/* CREATE */
+
+func (model MaintenanceWindowModel) toAPICreateRequest(ctx context.Context) (kbapi.PostMaintenanceWindowJSONRequestBody, diag.Diagnostics) {
+	body := kbapi.PostMaintenanceWindowJSONRequestBody{
+		Enabled: model.Enabled.ValueBoolPointer(),
+		Title:   model.Title.ValueString(),
+	}
+
+	body.Schedule.Custom.Duration = model.CustomSchedule.Duration.ValueString()
+	body.Schedule.Custom.Start = model.CustomSchedule.Start.ValueString()
+
+	if !model.CustomSchedule.Timezone.IsNull() && !model.CustomSchedule.Timezone.IsUnknown() {
+		body.Schedule.Custom.Timezone = model.CustomSchedule.Timezone.ValueStringPointer()
+	}
+
+	customRecurring, diags := model.CustomSchedule.Recurring.toAPIRequest(ctx)
+	body.Schedule.Custom.Recurring = customRecurring
+	body.Scope = model.Scope.toAPIRequest()
+
+	return body, diags
+}
+
+/* READ */
+
+func (model *MaintenanceWindowModel) fromAPIReadResponse(ctx context.Context, data *kbapi.GetMaintenanceWindowIdResponse) diag.Diagnostics {
+	if data == nil {
+		return nil
+	}
+
+	var diags = diag.Diagnostics{}
+	var response = &ResponseJson{}
+
+	if err := json.Unmarshal(data.Body, response); err != nil {
+		diags.AddError(err.Error(), "cannot unmarshal GetMaintenanceWindowIdResponse")
+		return diags
+	}
```

Comment:
Can we get rid of the intermediate types now we're only ever populating this from the `GetMaintenanceWindowIdResponse`? I'm happy if you still see value in this step.